tools/automation.py

```diff
@@ -252,45 +296,37 @@ def get_all_mentors_in_yml_format(xlsx_file_path, skip_rows=0):
 
 def get_new_mentors_in_yml_format(yml_file_path, xlsx_file_path, skip_rows=1):
     """
-    Read just new mentors from Excel sheet
+    Read just new mentors from Excel sheet:
      - start reading xlsx Mentors from the row 1 (from the date 03/04/2024)
-     - find diff. between existing yml and xlsx
-    Prepare data for writing to yaml file.
+     - find diff. between current mentors.yml and xlsx table
     """
-    # list of dict
-    mentors = []
-
-    # Get mentors' names and indexes from yml file
-    mentors_yml_dict = read_yml_file(yml_file_path)
-
     df_mentors = pd.read_excel(xlsx_file_path, sheet_name=SHEET_NAME, skiprows=skip_rows)
 
     logging.info(f"Excel mentors: {len(df_mentors)}")
 
-    if mentors_yml_dict:
-        mentors_names_yml = [sub['name'].lower() for sub in mentors_yml_dict]
-        mentors_indexes = [sub['index'] for sub in mentors_yml_dict]
+    # Get current mentors' data in mentors.yml
+    df_yml = get_yml_data(yml_file_path)
 
-        logging.info(f"Mentors yml total: {len(mentors_names_yml)}")
+    # list of dict
+    mentors = []
 
+    if not df_yml.empty:
         # Highest index is used as the reference point from which
         # new indexes are calculated
-        new_index = max(mentors_indexes) + 1
+        new_index = df_yml['Index'].max().item() + 1
```

Comment:
Again, the comment is not necessary, you could rename the variable to `next_index` for example. And the code is clear about what is doing ;)

Reply:
@nverbic please address the comments in another pr.